lib/matplotlib/axes/_base.py

```diff
@@ -514,6 +514,8 @@ def _plot_args(self, tup, kwargs, return_kwargs=False):
         ncx, ncy = x.shape[1], y.shape[1]
         if ncx > 1 and ncy > 1 and ncx != ncy:
             raise ValueError(f"x has {ncx} columns but y has {ncy} columns")
+        if not ncx or not ncy:
```

Comment:
I have a slight preference for 
```suggestion
        if ncx == 0 or ncy == 0:
```

as it is more explicit about what we care about catching and people tend to have an easier time understanding non-negated logic (which is an example of what I just said is hard to understand ;) )